usl_models/usl_models/flood_ml/model.py

```diff
@@ -0,0 +1,129 @@
+"""Flood model definition."""
+
+from typing import Any
+
+import tensorflow as tf
+from tensorflow.keras import layers
+
+
+class FloodConvLSTM(tf.keras.Model):
+    """Flood ConvLSTM model.
+
+    The architecture is an autoregressive ConvLSTM. Spatiotemporal and
+    geospatial features are passed through initial CNN blocks for feature
+    extraction, then concatenated with temporal inputs. The combined inputs are
+    then passed into ConvLSTM and TransposeConv layers to output a map of flood
+    predictions.
+
+    The spatiotemporal inputs are "initial condition" flood maps, with previous
+    flood predictions being fed back into the model for future predictions.
+    This creates the autoregressive loop. We define a maximum number
+    N_FLOOD_MAPS of flood maps to use as inputs.
+
+    Architecture diagram: https://miro.com/app/board/uXjVKd7C19U=/.
+    """
+
+    def __init__(self, params: dict[str, Any]):
+        """Creates the ConvLSTM model.
+
+        Args:
+            params: A dict of tunable model parameters.
+        """
+        super().__init__()
+
+        self.params = params
+
+        # Spatiotemporal CNN
+        st_cnn_params = {"strides": 2, "padding": "same", "activation": "relu"}
+        st_cnn_layers = [
+            layers.Conv2D(16, 5, **st_cnn_params),
+            layers.MaxPool2D(pool_size=2, strides=1, padding="same"),
+            layers.Conv2D(64, 5, **st_cnn_params),
+            layers.MaxPool2D(pool_size=2, strides=1, padding="same"),
+        ]
+        self.st_cnn = tf.keras.Sequential(
+            [layers.TimeDistributed(layer) for layer in st_cnn_layers]
```

Comment:
It looks like Sequential will accept a generator, so you can avoid materializing that list by removing those brackets
```python
 self.st_cnn = tf.keras.Sequential(
    layers.TimeDistributed(layer) for layer in st_cnn_layers
)

Reply:
Done.